dev-tools/composer.lock

```diff
@@ -2356,16 +2356,16 @@
         },
         {
             "name": "phpstan/phpstan",
-            "version": "1.10.3",
```

Comment:
pls bump composer.json too, otherwise there is higher possibility someone will regenerate lock file and move to older version of phpstan again.

(we still want to have lock file for deep dependencies)

Reply:
In the future it would be better to have Renovate or Dependabot to do it automatically (the former has various strategies, can bump json too).

Personally I think it's better to bump constraint in JSON only when it's really needed (using feature from specific version, so constraint ensures it will be installed). Bumping constraints causes lock's content hash change and it leads to conflicts.

This is my opinion, but I'll do as you wish 😉.

Reply:
using or not using bot is irrelevant of whether we should update 1 or 2 files

for using library, like symfony/console, I agree, you put lowest possible in json file and highest possible in lock file [or not have lock at all]

yet here, we are not using library, we install phpstan app that we want to run on given, specific version Imagine we say we want all errors to be reported, and claim version ^5.6.7 in json file, while 5.99.99 in lock file.
if someone would regenerate lock file due to eagerness to install some other lib, that is incompatible with 5.99.99, they could end up installing 5.10.0, even if we wanted to check all errors from 5.99.99 already

Reply:
I understand, but I don't see the difference. For example, if some kind of error we had in baseline would be reported in specific PHPStan version, then we should bump it in `composer.json`, otherwise it may lead to "ignore rule not matched" reports and PHPStan failure when version is downgraded (which would explicitly tell something need to be fixed, that's why I added `reportUnmatchedIgnoredErrors: true`). If minimal required version and latest version reports the same set of issues in the codebase, then it does not matter which is installed. On the other hand, bumping minimal required version could lead to conflicts with other packages' requirements, so why making requirements resolution more complex? 🙂

Reply:
i see the value of `reportUnmatchedIgnoredErrors: true` in this specific case.
let's move to other dev-tool, random example: phpmd. keeping vX.0 in json file, but vX.9 in lock file allows one to easily reduce version level in lock file without officially claiming so - eg by installing other tool that is not compatible with vX.9 for whatever reason. and then, our CI would stop reporting some level of issues and if one would realize and manually upgrade to vX.9 again, it may reveal the issues that should never be introduced.
and there is no baseline for exceptions.

and so, we should have same approach for all dev-tools